File: src/features/datasets/domain/dataset.ts
export interface DatasetRecord {
  id: string;
  title: string;
  description: string;
  category: string;
  tags: string[] | null;
  fileKey: string;
  fileName: string;
  fileSize: number;
  fileType: string;
  downloadCount: number | null;
  isPublic: boolean | null;
  publishedAt: Date | null;
  updatedAt: Date | null;
  createdAt: Date | null;
  source: string | null;
  license: string | null;
  version: string | null;
  coverage: string | null;
  format: string;
  keywords: string[] | null;
  methodology: string | null;
}

export interface DatasetFilterInput {
  category?: string;
  search?: string;
  tags?: string[];
  format?: string;
  page: number;
  limit: number;
}

export interface AdminDatasetFilterInput extends DatasetFilterInput {
  includePrivate: boolean;
}

export interface DatasetUploadInput {
  title: string;
  description: string;
  category: string;
  tags: string[];
  source?: string;
  license: string;
  version: string;
  coverage?: string;
  format: string;
  keywords: string[];
  methodology?: string;
  isPublic: boolean;
  fileKey: string;
  fileName: string;
  fileSize: number;
  fileType: string;
}

export interface DatasetUpdateInput {
  id: string;
  title?: string;
  description?: string;
  category?: string;
  tags?: string[];
  source?: string;
  license?: string;
  version?: string;
  coverage?: string;
  format?: string;
  keywords?: string[];
  methodology?: string;
  isPublic?: boolean;
}

export interface DatasetCategoryStat {
  category: string;
  count: number;
}

File: backend/internal/domain/entity/dataset.go
package entity

import "time"

type Dataset struct {
	ID            string     `json:"id"`
	Title         string     `json:"title"`
	Description   string     `json:"description"`
	Category      string     `json:"category"`
	Tags          []string   `json:"tags,omitempty"`
	FileKey       string     `json:"fileKey"`
	FileName      string     `json:"fileName"`
	FileSize      int        `json:"fileSize"`
	FileType      string     `json:"fileType"`
	DownloadCount int        `json:"downloadCount"`
	IsPublic      bool       `json:"isPublic"`
	PublishedAt   *time.Time `json:"publishedAt,omitempty"`
	Source        *string    `json:"source,omitempty"`
	License       string     `json:"license"`
	Version       string     `json:"version"`
	Coverage      *string    `json:"coverage,omitempty"`
	Format        string     `json:"format"`
	Keywords      []string   `json:"keywords,omitempty"`
	Methodology   *string    `json:"methodology,omitempty"`
	CreatedAt     time.Time  `json:"createdAt"`
	UpdatedAt     time.Time  `json:"updatedAt"`
}
